### ISSUE TITLE
Overlaying textures of different sizes.

### ISSUE THREAD

**Issue by pithydon** [minetest/minetest#4891]
I think Minetest should scale the smaller textures up, but this is not always the case.
To make things worse the current behavior is inconsistent.
![screenshot_20161211_135847](https://cloud.githubusercontent.com/assets/18400998/21083303/952df0b8-bfaa-11e6-9a38-70fb996e1188.jpg)

**Comment by tobyplowy** [comment]
@pithydon +1 all the way!

**Comment by sofar** [comment]
I'm having a hard time understanding the problem. How would I reproduce this issue? What code did you use to generate the above screenshot?

**Comment by pithydon** [comment]
It's my fossils mod with lucid texture pack.
I think the behavior may be more consistent then it first seems.
The bug happens when a smaller minetest generated texture is overlaid a larger texture.
```
"default_stone.png^(fossils_theropod_skull_s.png^[transformfx)"
```
It does not happen when the smaller texture is a file.
```
"default_stone.png^fossils_theropod_skull_s.png"
```

**Comment by pithydon** [comment]
@paramat can you explain how this is a "feature request" and not a bug?

**Comment by DS-Minetest** [comment]
isnt there a`[resize:<w>x<h>`? why dont you use it?
and this also happened in the past with texture packs, that edited the stone texture but not the mineral

**Comment by pithydon** [comment]
```[resize:<w>x<h>``` would resize the texture to a hardcoded size.
Not the best for texture packs either.
I could use ```[resize:512x512``` then only people with slow computers would notice.

**Comment by SmallJoker** [comment]
Fixed by #4016
